### PR TITLE
Addresses #720 and #721 in the new explainer

### LINKED ISSUE #720: Timing of user consent requests
**[Disclaimer: This issue is one of several being filed to capture discussions that began either on #638, on #689, or at the most recent F2F]**

When getting user consent is required to protect sensitive information, there appear to be three options for when the User Agent can ask for the consent. (Note, this is not the issue for discussing the duration of consent. That can be found in a #721)

1. At the time a session is created
1. At the time an API is invoked which developer intent to use information that requires user consent (e.g. requestReferenceSpace())
1. Some combination of the above depending on the thing consent is being requested for

From what I’ve been able to gather, the following questions need to be answered before we’ll be able to come to a conclusive design:

* Do all User Agents need to align on a single option from the list above? This is really a question of principle rather than how developers would use such an API in practice. In other words, if it would be possible for a developer to write a single predictable code path that would allow User Agents the choice of when to prompt, is that acceptable? Desirable? Concerning?
* Does requesting consent in the middle of an immersive session require the use of Trusted Immersive UI (see #718)? IIRC, John mentioned this might be necessary to ensure end users don’t get confused or tricked by a website spoofing Trusted Immersive UI if they’ve gotten used to consent prompts not being Trusted. (Note, resolving this bullet point may cause a dependency on #719 )

Are there other considerations I’ve missed?

### LINKED ISSUE #721: Duration of user consent
**[Disclaimer: This issue is one of several being filed to capture discussions that began either on #638, on #689, or at the most recent F2F]**

When user consent is required and granted, there are three options I’m aware of for how long user consent can last. (Note, this is not the issue for discussing what triggers UAs to request consent. That can be found in #720 )

* Session, which means each new XRSession would require new prompting
* Browsing-context, which is defined in the HTML Standard and implies user consent would be cumulative when multiple are XRSessions created in the same browsing context.
* Origin, which is defined in the HTML Standard and implies user consent would be cumulative for each XRSession created by that origin

Are there any other options I’ve left out? Is this a decision which can be left up to each User Agent or is this something that must be consistent across all? Are there minimum requirements for the duration based on the type of data being secured?

### PR DESCRIPTION
**Pull request by NellWaliczek**
As mentioned in #733, here's the PR that fixes #720 and fixes #721.  This PR doesn't include spec text just yet. (Yes, that means I'm absolutely breaking my rule about no explainer changes without spec text changes.)  I'm currently working on a follow-up PR that will lay in spec text but it is intertwined with the solution to another issue that is hopefully close to resolution as well.  In the meanwhile, this PR aims ensure we're on the same page regarding desired behavior of explicit consent requests and consent duration.

### CODE REVIEW

**Comment by joshmarinacci** [comment]
Looks good to me. Thanks Nell!

**Comment by NellWaliczek** [comment]
Rebased on the latest #733 and added missing information about origns in the section on duration of consent.  @toji  and @johnpallett can you also take a look?

**Comment by NellWaliczek** [comment]
Rebased on master and ready to go as soon as I've gotten sign off

File: privacy-security-explainer.md [hunk 1]
```diff
@@ -36,13 +36,10 @@ It is often necessary to be sure of user intent before exposing sensitive inform
...
-
-### Timing of XR-related consent prompts
-**TODO** Fill this in with what was agreed up on in [#720](https://github.com/immersive-web/webxr/issues/720).
+It is often useful to get explicit consent from the user before exposing sensitive information. When gathering explicit user consent, User Agents present an explanation of what is being requested and provide users the option to decline. Requests for user consent can be presented in many visual forms based on the features being protected and User Agent choice. While often associated with the [Permissions API](https://www.w3.org/TR/permissions/), the concept of user consent does not have exact overlap. If sensitive data is protected by explicit consent and will be used during an `XRSession`, it is strongly recommended that User Agents display the associated consent prompt prior to the session being created.
```
Comment (toji) [inline privacy-security-explainer.md]: I assume the plan to revisit this once we start working on a 'deferred' feature category, and this text is only intended to apply to the 'required' and 'optional' categories that we've discussed?

File: privacy-security-explainer.md [hunk 2]
```diff
@@ -36,13 +36,10 @@ It is often necessary to be sure of user intent before exposing sensitive inform
...
 
 ### Duration of consent
-**TODO** Fill this in with what was agreed up on in [#721](https://github.com/immersive-web/webxr/issues/721).
+It is strongly recommended that once consent is granted, either implicitly or explicitly, for a specific [origin](https://html.spec.whatwg.org/multipage/origin.html) that this consent persist until the [browsing context](https://html.spec.whatwg.org/multipage/browsers.html#browsing-context) has ended. User agents may also choose to allow consent to persist for all subsequent browsing contexts of that origin. Caution is advised when doing do, especially when done in response to implied consent.
```
Comment (toji) [inline privacy-security-explainer.md]: My impression previously is that we wanted access to features consent was granted for to be scoped to session lifetime. (As you stated here, the consent itself could persist between sessions in the same browsing context.) Where would be the appropriate place to make that distinction? In this section, or is there another stub that's earmarked for it?

File: privacy-security-explainer.md (hunk 1 above)
Comment (NellWaliczek) [reply privacy-security-explainer.md]: Yes.  Filed #740.

File: privacy-security-explainer.md (hunk 2 above)
Comment (NellWaliczek) [reply privacy-security-explainer.md]: I'm not sure what you're referring to?  This text was intended to reflect my understanding of the consensus reached in https://github.com/immersive-web/webxr/issues/721#issuecomment-504614488 and https://github.com/immersive-web/webxr/issues/721#issuecomment-505224211.  @johnpallett and @avadacatavra, did I misunderstand?

Comment (toji) [reply privacy-security-explainer.md]: You know what, it's not actually relevant at this point. Previous conversations were around things like "granting a session access to the camera shouldn't grant that access for the entire page" but that's not something that we're considering at this stage. The features that we ARE discussing (primarily reference space access) are naturally session-bound anyway, so it's a moot point. This is fine.

[the remark this answers is not in the document]
Comment (NellWaliczek) [reply privacy-security-explainer.md]: Thanks, John!  That's a great suggestion.  And thanks for the exact text.  @avadacatavra, does that work for you?  I'd like to merge this one today if possible.

**Comment by NellWaliczek** [comment]
Ok, I haven't heard back from @avadacatavra yet, but any changes she may have will probably be small in scope, so I'm going to go ahead and merge this one.

**Pull request merged**